### DISCUSSION TITLE
Announcement channel from message mentions

### DISCUSSION THREAD

**Discussion by kiwisap** [discord-jda/JDA#2027] [Questions and Help] [answered]
Hey,

So I was creating this feature in my bot where I could mention a channel at the top of my message, to send the rest of the message in the mentioned channel. This is all working, but somehow not for Announcement (News) channels. I was wondering if there is a way to get such a channel from a mention.

Some code to help you understand:
```
@Override
public void onMessageReceived(@NotNull MessageReceivedEvent event) {
    // Guild check etc etc...
    
    List<IMentionable> mentionedChannels = event.getMessage().getMentions(Message.MentionType.CHANNEL);
    if (mentionedChannels.isEmpty()) return;

    IMentionable mentionedChannel = mentionedChannels.get(0);
    if (!(mentionedChannel instanceof GuildMessageChannel)) return;

    GuildMessageChannel channel = (GuildMessageChannel) mentionedChannel;
}
```

When mentioned any textchannel `mentionedChannels` is not empty. When mentioned any newschannel `mentionedChannels` is always empty, somehow it doesn't get newschannels.

Any help would be appreciated!

Thanks in advance! :)

**Comment by RedcodesDev** [comment, marked as answer]
This looks like a bug in the JDA to me. I tested it with JDA Version 5.0.0-alpha.6, and it always returns an empty list if I mention a news channel. My guess is that they forgot to add NewsChannel to the MentionType.CHANNEL because the whole channel hierarchy was redone in the 5.0.0 alphas. I also tried the code in v4, and it worked as expected.

**Reply by RedcodesDev** [reply]
I've created a Bugreport in #2034

**Reply by kiwisap** [reply]
Allright thanks 🙂
